### ISSUE TITLE
Allow ifn for column-selector in replace-missing

### ISSUE THREAD

**Issue by kimim** [techascent/tech.ml.dataset#182]
I found it is convenient to allow `ifn` for `column-selector` in `replace-missing`, for example:

```clojure
(deftest replace-missing-selector-fn
  (let [ds (ds/->dataset {:a [nil nil 2 4]
                          :b [nil nil 4 6]
                          :c [nil nil "A" "B"]})
        ds-replaced (-> ds
                        (ds/replace-missing cf/numeric :value dfn/mean)
                        (ds/replace-missing cf/categorical :value "C"))]
    (is (= [3 3 2 4] (vec (ds-replaced :a))))
    (is (= [5 5 4 6] (vec (ds-replaced :b))))
    (is (= ["C" "C" "A" "B"] (vec (ds-replaced :c))))))
```

This can be done by modify [ `replace-missing` ](https://github.com/techascent/tech.ml.dataset/blob/309dc98c52380791a6911ee8f024a9503643fe79/src/tech/v3/dataset/missing.clj#L199)

```clojure
(defn replace-missing
  ...
  ([ds columns-selector strategy value]
   (let [strategy (or strategy :mid)
         row-cnt (ds-base/row-count ds)
         selected (if (fn? columns-selector)  ;; <---
                    (columns-selector ds)
                    (ds-base/select-columns ds columns-selector))]
     (->> selected
```

**Comment by cnuernber** [comment]
Yes, great idea.

**Comment by kimim** [comment]
Shall I create a PR for this issue?

**Comment by cnuernber** [comment]
Yes please, that would be great.

**Comment by kimim** [comment]
fix by #183

**Comment by cnuernber** [comment]
Included in newest version!
